app/src/main/java/com/duckduckgo/app/notification/NotificationSender.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.notification
+
+import android.content.Context
+import androidx.core.app.NotificationManagerCompat
+import com.duckduckgo.app.notification.db.NotificationDao
+import com.duckduckgo.app.notification.model.Notification
+import com.duckduckgo.app.notification.model.SchedulableNotification
+import com.duckduckgo.app.pixels.AppPixelName
+import com.duckduckgo.app.statistics.pixels.Pixel
+import timber.log.Timber
+
+interface NotificationSender {
+    suspend fun sendNotification(context: Context, notification: SchedulableNotification)
+}
+
+class AppNotificationSender(
+    private val pixel: Pixel,
+    private val manager: NotificationManagerCompat,
+    private val factory: NotificationFactory,
+    private val notificationDao: NotificationDao
+) : NotificationSender {
+
+    override suspend fun sendNotification(context: Context, notification: SchedulableNotification) {
```

Comment:
NIT: could we have context as param of `AppNotificationSender` constructor? it seems it could be injected but not sure if I'm missing something.

Reply:
Makes sense! Done!